src/main/java/org/sonar/plugins/stash/StashIssueReportingPostJob.java

```diff
@@ -21,96 +24,115 @@
-              }
-
-              // if requested, reset all comments linked to the pull-request
-              if (config.resetComments()) {
-                  stashRequestFacade.resetComments(stashProject, repository, stashPullRequestId,
-                          diffReport, stashUser, stashClient);
-              }
-
-              boolean canApprovePullrequest = config.canApprovePullRequest();
-              if (canApprovePullrequest) {
-                  stashRequestFacade.addPullRequestReviewer(stashProject, repository, stashPullRequestId,
-                          stashCredentials.getLogin(), stashClient);
-              }
-
-              // if threshold exceeded, do not push issue list to Stash
-              if (issueReport.countIssues() >= issueThreshold) {
-                  LOGGER.warn("Too many issues detected ({}/{}): Issues cannot be displayed in Diff view",
-                          issueReport.countIssues(), issueThreshold);
-              } else {
-                  stashRequestFacade.postCommentPerIssue(stashProject, repository, stashPullRequestId, sonarQubeURL,
-                          issueReport, diffReport, stashClient);
-              }
-
-              stashRequestFacade.postAnalysisOverview(stashProject, repository, stashPullRequestId, sonarQubeURL,
-                      issueThreshold, issueReport, stashClient);
-
-              // if no new issues, plugin approves the pull-request
-              if (canApprovePullrequest && issueReport.countIssues() == 0) {
-                  stashRequestFacade.approvePullRequest(stashProject, repository, stashPullRequestId,
-                          stashCredentials.getLogin(), stashClient);
-
-              } else if (canApprovePullrequest && issueReport.countIssues() != 0) {
-                  stashRequestFacade.resetPullRequestApproval(stashProject, repository, stashPullRequestId,
-                          stashCredentials.getLogin(), stashClient);
-              }
+    StashClient stashClient = null;
+    try {
+      boolean notifyStash = config.hasToNotifyStash();
+      if (notifyStash) {
+        
+        SonarQubeIssuesReport issueReport = stashRequestFacade.extractIssueReport(projectIssues, inputFileCache);
+          
+        int issueThreshold = stashRequestFacade.getIssueThreshold();
+        String sonarQubeURL = config.getSonarQubeURL();
+          
+        // Stash MANDATORY options
+        String stashURL = stashRequestFacade.getStashURL();
+        String stashProject = stashRequestFacade.getStashProject();
+        String repository = stashRequestFacade.getStashRepository();
+        String stashPullRequestId = stashRequestFacade.getStashPullRequestId();
+          
+        int stashTimeout = config.getStashTimeout();
+          
+        StashCredentials stashCredentials = stashRequestFacade.getCredentials();
+
+        stashClient = new StashClient(stashURL, stashCredentials, stashTimeout, config.getSonarQubeVersion());

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "StashClient". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2095)